x/concentrated-liquidity/lp_test.go

```diff
@@ -481,9 +543,26 @@ func (s *KeeperTestSuite) TestWithdrawPosition() {
 			incentivesBalanceAfterCollect := s.App.BankKeeper.GetAllBalances(ctx, pool.GetIncentivesAddress())
 			ownerBalancerAfterCollect := s.App.BankKeeper.GetAllBalances(ctx, owner)
 
+			expectedOwnerBalanceDelta := expectedPoolBalanceDelta.Add(expectedIncentivesClaimed...)
+			actualOwnerBalancerDelta := ownerBalancerAfterCollect.Sub(ownerBalancerBeforeCollect)
+
+			actualIncentivesClaimed := incentivesBalanceBeforeCollect.Sub(incentivesBalanceAfterCollect)
+
 			s.Require().Equal(expectedPoolBalanceDelta.String(), poolBalanceBeforeCollect.Sub(poolBalanceAfterCollect).String())
-			s.Require().Equal(expectedPoolBalanceDelta.Add(expectedIncentivesClaimed...).String(), ownerBalancerAfterCollect.Sub(ownerBalancerBeforeCollect).String())
-			s.Require().Equal(expectedIncentivesClaimed.String(), incentivesBalanceBeforeCollect.Sub(incentivesBalanceAfterCollect).String())
+
+			// TODO: Investigate why full range liquidity positions are slightly under claiming incentives here
+			errTolerance := osmomath.ErrTolerance{
+				AdditiveTolerance: sdk.NewDec(3),
+				RoundingDir:       osmomath.RoundDown,
+			}
+
+			for _, coin := range expectedOwnerBalanceDelta {
+				errTolerance.Compare(expectedOwnerBalanceDelta.AmountOf(coin.Denom), actualOwnerBalancerDelta.AmountOf(coin.Denom))
+			}
+
+			for _, coin := range expectedIncentivesClaimed {
+				errTolerance.Compare(expectedIncentivesClaimed.AmountOf(coin.Denom), actualIncentivesClaimed.AmountOf(coin.Denom))
+			}
```

Comment:
We need to make sure two things here:
1. The likelihood of false positives is minimized by checking that `expectedOwnerBalanceDelta` is not empty when applicable
2. Validating the result of `errTolerance.Compare`. Currently, that is a no-op

Reply:
Good catch on the later. Fixed both here https://github.com/osmosis-labs/osmosis/pull/4883/commits/445684ff7cd8c677997942553799d22279e0aceb